src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -173,11 +174,30 @@ def _update_command_table_from_extensions():
                         logger.warning("Unable to load extension '%s'. Use --debug for more information.", ext_name)
                         logger.debug(traceback.format_exc())
 
+        def _get_extension_block_filters(mod_loaders):
+            filter_func_name = 'module_block_extension'
+            return [getattr(m, filter_func_name) for m in mod_loaders if callable(getattr(m, filter_func_name, None))]
+
+        def _wrap_block_extension_func(func, ext):
+            """ Wrapper method to handle centralization of log messages for extension filters """
+            res = func(ext)
+            should_block = res
+            reason = "Use --debug for more information."
+            if isinstance(res, tuple):
+                should_block, reason = res
+            if should_block:
```

Comment:
Should we include a check for `should_block` being a bool? If someone returns something random in their blocking function this might lead to weird behavior.

Reply:
I've casted it to a bool.